examples/wallet/app/components/RestoreWalletFromPrivateKey.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+import React, { useState } from 'react';
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+import Container from 'react-bootstrap/Container';
+import Row from 'react-bootstrap/Row';
+import typeof { setAccount as SetAccount } from '../actions/account';
+
+type Props = {
+  setAccount: SetAccount
+};
+
+// FIXME: this has no error handling, neither while parsing the address
+// nor when fetching the balance.
+export default ({ setAccount }: Props) => {
+  const handleSubmitCreateSpending = function handleSubmitCreateSpending(
+    event
+  ) {
+    event.preventDefault();
+    if (mustCreateSpendingPassword) {
+      if (checkValidSpendingPassword(password, confirmPassword)) {
+        setAccount(newPrivateKey, password);
+      }
+    } else {
+      setAccount(newPrivateKey, '');
+    }
+  };
+
+  const handleCheckCreateSpendingPassword = function handleCheckCreateSpendingPassword(
+    evt
+  ) {
+    setMustCreateSpendingPassword(evt.target.checked);
+    setHiddenSpendingPassword(!evt.target.checked);
+  };
+
+  const checkValidSpendingPassword = function checkValidSpendingPassword(
+    pass,
+    confirmation
+  ) {
```

Comment:
I find it confusing that the function both calls the state setters and returns the value of the check, I'd reccommend having it only return the value of the check, and call the setters in the calling context.